### DISCUSSION TITLE
Dynamical matrix does not satisfy the acoustic sum rule after several minimization steps

### DISCUSSION THREAD

**Discussion by Kurufinve** [SSCHAcode/python-sscha#345] [Q&A]
Dear SSCHA developers and users,

Sometimes, during the SSCHA minimization at the one of the SSCHA relaxation steps (i.e. with ensemble produced not from initial dynamical matrix)  I face with this error:

```
Immaginary frequencies found! Redoing the step.
Step too large (scalar = 0.017684261306056173 | kl_ratio = 0.0), reducing to 0.00039018442352848536

ERROR WHILE UPDATING THE WEIGHTS

Error, one dynamical matrix does not satisfy the acoustic sum rule.
    If this problem arises on a sscha run,
    it may be due to a gradient that violates the sum rule.
    Please, be sure you are not using a custom gradient function.

DETAILS OF ERROR:
    Number of translatinal modes in the original dyn = 3
    Number of translational modes in the target dyn = 1
    (They should be both 3)
```

Does anyone knows, how to reduce the probability to face with this error?

**Comment by mesonepigreco** [comment]
It seems something very pathological is occurring here. The code tells you that ``Immaginary frequencies found! Redoing the step.`` This already is a bad sign. 
Then, ``kl_ratio = 0.0``, it means that with one step it went completely away from the starting point. The absence of translational modes in the target dynamical matrix (as pointed) means that the final dynamical matrix is completely wrong. This should not occur. 
If it is the case, try using 

```
minim.root_representation = "sqrt" # or "root4"
```

It should improve the stability of the minimization and prevent the occurrence of imaginary frequencies.
